packages/babel-types/src/clone/cloneNode.ts

```diff
@@ -5,19 +5,21 @@ import { isFile, isIdentifier } from "../validators/generated";
 const has = Function.call.bind(Object.prototype.hasOwnProperty);
 
 // This function will never be called for comments, only for real nodes.
-function cloneIfNode(obj, deep, withoutLoc) {
+function cloneIfNode(obj, deep, withoutLoc, commentsCache) {
   if (obj && typeof obj.type === "string") {
-    return cloneNode(obj, deep, withoutLoc);
+    return cloneNodeInternal(obj, deep, withoutLoc, commentsCache || new Map());
   }
 
   return obj;
 }
 
-function cloneIfNodeOrArray(obj, deep, withoutLoc) {
+function cloneIfNodeOrArray(obj, deep, withoutLoc, commentsCache) {
+  commentsCache ??= new Map();
+
   if (Array.isArray(obj)) {
-    return obj.map(node => cloneIfNode(node, deep, withoutLoc));
+    return obj.map(node => cloneIfNode(node, deep, withoutLoc, commentsCache));
```

Comment:
We should use `cloneIfNodeInternal` here, and avoid adding the new parameter to `cloneIfNode` (since it's part of the public API, and this is an internal implementation detail)

Reply:
Ops, I was reading the code wrong.